### ISSUE TITLE
Confirming edit of roof path edit with button causes segfault

### ISSUE THREAD

**Issue by sboddy** [IfcOpenShell/IfcOpenShell#5750]
Steps to reproduce:
1. Create a new file based on IFC4 demo.
2. Add an empty and change it to an IfcRoof element (no type needed).
3. Go to the parametric geometry roof part in the sidebar and hit the plus icon
4. Click the squiggly arrow button to edit the path
5. Click the tick button to confirm edit (even though nothing was changed)

Segfault! (see below)

Happens every time for me. Curiously, if I use tab to exit out of edit path mode, it doesn't segfault.

```
GNU gdb (Ubuntu 12.1-0ubuntu1~22.04.2) 12.1
Copyright (C) 2022 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<https://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from ./blender...
(gdb) run
Starting program: /home/steve/Software/blender-git/build_linux/bin/blender 
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
[New Thread 0x7fffe2fff600 (LWP 3782879)]
[New Thread 0x7fffc88fe600 (LWP 3782880)]
[New Thread 0x7fffc80fd600 (LWP 3782881)]
[New Thread 0x7fffc78fc600 (LWP 3782882)]
[New Thread 0x7fffc70fb600 (LWP 3782883)]
[New Thread 0x7fffc68fa600 (LWP 3782884)]
[New Thread 0x7fffc60f9600 (LWP 3782885)]
[New Thread 0x7fffc58f8600 (LWP 3782886)]
[New Thread 0x7fffc50f7600 (LWP 3782887)]
[New Thread 0x7fffbc7f2600 (LWP 3782888)]
[New Thread 0x7fffb79ff600 (LWP 3782889)]
[New Thread 0x7fffbbff1600 (LWP 3782890)]
[New Thread 0x7fffbb1ff600 (LWP 3782891)]
[New Thread 0x7fffba9fe600 (LWP 3782892)]
[New Thread 0x7fffb98b2600 (LWP 3782893)]
[New Thread 0x7fffb8bff600 (LWP 3782894)]
[New Thread 0x7fffa55e0600 (LWP 3782895)]
[New Thread 0x7fff9b924600 (LWP 3782896)]
[New Thread 0x7fff9abff600 (LWP 3782897)]
[New Thread 0x7fff9a3fe600 (LWP 3782898)]
[New Thread 0x7fff99bfd600 (LWP 3782899)]
[New Thread 0x7fff993fc600 (LWP 3782900)]
[New Thread 0x7fff98bfb600 (LWP 3782901)]
[New Thread 0x7fff983fa600 (LWP 3782902)]
[New Thread 0x7fff97bf9600 (LWP 3782903)]
[New Thread 0x7fff973f8600 (LWP 3782904)]
[New Thread 0x7fff8f7ff600 (LWP 3782908)]
[New Thread 0x7fff8effe600 (LWP 3782909)]
[New Thread 0x7fff8e3ff600 (LWP 3782910)]
[New Thread 0x7fff8d5ff600 (LWP 3782911)]
[New Thread 0x7fff8c7ff600 (LWP 3782912)]
[New Thread 0x7fff8bbfe600 (LWP 3782914)]
[New Thread 0x7fff8afff600 (LWP 3782915)]
[New Thread 0x7fff89dff600 (LWP 3782916)]
[New Thread 0x7fff895fe600 (LWP 3782917)]
[Detaching after vfork from child process 3782920]
[Detaching after vfork from child process 3782921]
[Detaching after vfork from child process 3782922]
[New Thread 0x7fff6b00c600 (LWP 3782926)]
[New Thread 0x7fff6ac0b600 (LWP 3782927)]
[New Thread 0x7fff6a1ff600 (LWP 3782928)]
[New Thread 0x7fff69bfe600 (LWP 3782929)]
[New Thread 0x7fff697fd600 (LWP 3782930)]
[New Thread 0x7fff68fff600 (LWP 3782931)]
[New Thread 0x7fff683ff600 (LWP 3782932)]
[New Thread 0x7fff669ff600 (LWP 3782933)]
[Thread 0x7fff669ff600 (LWP 3782933) exited]
[New Thread 0x7fff5dbff600 (LWP 3782935)]
[New Thread 0x7fff5d3fe600 (LWP 3782936)]
New cache was created.
[New Thread 0x7fff669ff600 (LWP 3783248)]
[New Thread 0x7fff5bfff600 (LWP 3783249)]
[Thread 0x7fff5bfff600 (LWP 3783249) exited]
[Thread 0x7fff669ff600 (LWP 3783248) exited]
[New Thread 0x7fff669ff600 (LWP 3783250)]
[New Thread 0x7fff5bfff600 (LWP 3783251)]
[Thread 0x7fff5bfff600 (LWP 3783251) exited]
[Thread 0x7fff669ff600 (LWP 3783250) exited]
[New Thread 0x7fff669ff600 (LWP 3783252)]
[Detaching after vfork from child process 3783253]
[Detaching after vfork from child process 3783255]
[Thread 0x7fff669ff600 (LWP 3783252) exited]
[New Thread 0x7fff669ff600 (LWP 3783264)]
[Thread 0x7fff669ff600 (LWP 3783264) exited]
[New Thread 0x7fff669ff600 (LWP 3783281)]
[New Thread 0x7fff5bfff600 (LWP 3783282)]
[Thread 0x7fff5bfff600 (LWP 3783282) exited]
[Thread 0x7fff669ff600 (LWP 3783281) exited]
[New Thread 0x7fff669ff600 (LWP 3783283)]
[New Thread 0x7fff5bfff600 (LWP 3783284)]
[Thread 0x7fff5bfff600 (LWP 3783284) exited]
[Thread 0x7fff669ff600 (LWP 3783283) exited]

Thread 1 "blender" received signal SIGSEGV, Segmentation fault.
0x0000000000de389a in CustomData_get_active_layer_index (data=data@entry=0x7fff608c6190, type=type@entry=CD_BM_ELEM_PYPTR) at /home/steve/Software/blender-git/blender/source/blender/blenkernel/intern/customdata.cc:2860
2860	  return (layer_index != -1) ? layer_index + data->layers[layer_index].active : -1;
(gdb) bt
#0  0x0000000000de389a in CustomData_get_active_layer_index(CustomData const*, eCustomDataType) (data=data@entry=0x7fff608c6190, type=type@entry=CD_BM_ELEM_PYPTR)
    at /home/steve/Software/blender-git/blender/source/blender/blenkernel/intern/customdata.cc:2860
#1  0x0000000000de4137 in CustomData_free_layer_active(CustomData*, eCustomDataType, int) (data=0x7fff608c6190, type=CD_BM_ELEM_PYPTR, totelem=0)
    at /home/steve/Software/blender-git/blender/source/blender/blenkernel/intern/customdata.cc:3392
#2  0x00000000011346ba in BM_data_layer_free(BMesh*, CustomData*, int) (bm=bm@entry=0x7fff608c6108, data=data@entry=0x7fff608c6190, type=type@entry=33)
    at /home/steve/Software/blender-git/blender/source/blender/bmesh/intern/bmesh_interp.cc:934
#3  0x0000000001723ab8 in bpy_bmesh_dealloc(BPy_BMesh*) (self=0x7fff5c7d3ba0) at /home/steve/Software/blender-git/blender/source/blender/python/bmesh/bmesh_py_types.cc:3726
#4  0x000000000080768e in Py_DECREF (op=<optimised out>) at ./Include/object.h:538
#5  Py_XDECREF (op=<optimised out>) at ./Include/object.h:602
#6  _PyFrame_Clear (frame=0x7fffb9002138) at Python/frame.c:141
#7  0x00000000007f1493 in _PyEvalFrameClearAndPop (frame=<optimised out>, tstate=0x8ca3a38 <_PyRuntime+166328>) at Python/ceval.c:6406
#8  _PyEval_EvalFrameDefault (tstate=<optimised out>, frame=0x7fffb9002098, throwflag=<optimised out>) at Python/ceval.c:2435
#9  0x00000000007f0697 in _PyEval_EvalFrame (throwflag=0, frame=0x7fffb9002020, tstate=0x8ca3a38 <_PyRuntime+166328>) at ./Include/internal/pycore_ceval.h:73
#10 _PyEval_Vector (tstate=0x8ca3a38 <_PyRuntime+166328>, func=<optimised out>, locals=0x0, args=0x7fff5c7eb298, argcount=<optimised out>, kwnames=<optimised out>) at Python/ceval.c:6434
#11 0x00000000017154ee in bpy_class_call(bContext*, PointerRNA*, FunctionRNA*, ParameterList*) (C=<optimised out>, ptr=<optimised out>, func=<optimised out>, parms=0x7fffffffd230)
    at /home/steve/Software/blender-git/blender/source/blender/python/intern/bpy_rna.cc:8823
#12 0x00000000016cd5e5 in rna_operator_exec_cb(bContext*, wmOperator*) (C=<optimised out>, op=0x7fff60ef69c8) at /home/steve/Software/blender-git/blender/source/blender/makesrna/intern/rna_wm.cc:1443
#13 0x0000000000ff180a in wm_operator_invoke(bContext*, wmOperatorType*, wmEvent const*, PointerRNA*, ReportList*, bool, bool)
     (C=C@entry=0x7fffe14c9748, ot=ot@entry=0x7fff720fa508, event=event@entry=0x7fffc0e88e08, properties=properties@entry=0x7fffffffd440, reports=reports@entry=0x0, use_last_properties=use_last_properties@entry=true, poll_only=false) at /home/steve/Software/blender-git/blender/source/blender/windowmanager/intern/wm_event_system.cc:1587
#14 0x0000000000ff1ec1 in wm_operator_invoke (use_last_properties=true, poll_only=false, reports=0x0, properties=0x7fffffffd440, event=0x7fffc0e88e08, ot=0x7fff720fa508, C=0x7fffe14c9748)
    at /home/steve/Software/blender-git/blender/source/blender/windowmanager/intern/wm_event_system.cc:1538
#15 wm_operator_call_internal(bContext*, wmOperatorType*, PointerRNA*, ReportList*, wmOperatorCallContext, bool, wmEvent const*)
    (C=0x7fffe14c9748, ot=0x7fff720fa508, properties=0x7fffffffd440, reports=0x0, context=WM_OP_INVOKE_REGION_WIN, poll_only=<optimised out>, event=0x7fffc0e88e08)
    at /home/steve/Software/blender-git/blender/source/blender/windowmanager/intern/wm_event_system.cc:1776
#16 0x0000000000ff2402 in wm_operator_call_internal (event=0x0, poll_only=false, context=WM_OP_INVOKE_REGION_WIN, reports=0x0, properties=0x7fffffffd440, ot=0x7fff720fa508, C=0x7fffe14c9748)
    at /home/steve/Software/blender-git/blender/source/blender/windowmanager/intern/wm_event_system.cc:1694
#17 WM_operator_name_call_ptr(bContext*, wmOperatorType*, wmOperatorCallContext, PointerRNA*, wmEvent const*) (C=C@entry=0x7fffe14c9748, ot=0x7fff720fa508, context=WM_OP_INVOKE_REGION_WIN, 
    context@entry=WM_OP_INVOKE_DEFAULT, properties=0x7fffffffd440, event=0x0, event@entry=0x7fff75ad8d60)
    at /home/steve/Software/blender-git/blender/source/blender/windowmanager/intern/wm_event_system.cc:1824
#18 0x0000000000ffab4f in WM_operator_name_call_ptr_with_depends_on_cursor(bContext*, wmOperatorType*, wmOperatorCallContext, PointerRNA*, wmEvent const*, char const*)
    (C=C@entry=0x7fffe14c9748, ot=<optimised out>, opcontext=WM_OP_INVOKE_DEFAULT, properties=<optimised out>, event=0x7fff75ad8d60, event@entry=0x0, drawstr=<optimised out>)
    at /home/steve/Software/blender-git/blender/source/blender/windowmanager/intern/wm_event_system.cc:2014
#19 0x0000000001814369 in ui_apply_but_funcs_after(bContext*) (C=0x7fffe14c9748) at /home/steve/Software/blender-git/blender/source/blender/editors/interface/interface_handlers.cc:1031
#20 0x00000000018257c9 in ui_handler_region_menu(bContext*, wmEvent const*, void*) (C=0x7fffe14c9748, event=0x7fff60d7e478)
    at /home/steve/Software/blender-git/blender/source/blender/editors/interface/interface_handlers.cc:11717
#21 0x0000000000ff65fd in wm_handler_ui_call (always_pass=false, event=0x7fff60d7e478, handler=0x7fffbd402ec8, C=0x7fffe14c9748)
    at /home/steve/Software/blender-git/blender/source/blender/windowmanager/intern/wm_event_system.cc:857
#22 wm_handlers_do_intern(bContext*, wmWindow*, wmEvent*, ListBase*) (C=0x7fffe14c9748, win=0x7fffc0ed9820, event=0x7fff60d7e478, handlers=<optimised out>)
    at /home/steve/Software/blender-git/blender/source/blender/windowmanager/intern/wm_event_system.cc:3384
#23 0x0000000000ff7e5d in wm_handlers_do(bContext*, wmEvent*, ListBase*) (C=C@entry=0x7fffe14c9748, event=event@entry=0x7fff60d7e478, handlers=handlers@entry=0x7fffc0ed9950)
    at /home/steve/Software/blender-git/blender/source/blender/windowmanager/intern/wm_event_system.cc:3503
#24 0x0000000000ff8943 in wm_event_do_handlers(bContext*) (C=C@entry=0x7fffe14c9748) at /home/steve/Software/blender-git/blender/source/blender/windowmanager/intern/wm_event_system.cc:4130
#25 0x0000000000fe9120 in WM_main(bContext*) (C=C@entry=0x7fffe14c9748) at /home/steve/Software/blender-git/blender/source/blender/windowmanager/intern/wm.cc:637
#26 0x000000000072f472 in main(int, char const**) (argc=1, argv=0x7fffffffdca8) at /home/steve/Software/blender-git/blender/source/creator/creator.cc:596
```
System details:
```
os: Linux
os_version: #134-Ubuntu SMP Fri Sep 27 20:20:17 UTC 2024
python_version: 3.11.7
architecture: ('64bit', 'ELF')
machine: x86_64
processor: x86_64
blender_version: 4.2.3 LTS
bonsai_version: 0.8.1-alpha241113
bonsai_commit_hash: 099703f
last_actions: 
last_error: None
```

**Comment by Andrej730** [comment]
@sboddy Couldn't reproduce this on Windows. Duplicate of #5609? If it is duplicate it's upstream issue that's fixed in Blender 4.2.4 and Blender 4.3, please try to reproduce the issue using those versions (https://builder.blender.org/download/daily/)

**Comment by sboddy** [comment]
Doh! You're right. I forgot I tested #5609 with the edge install. The 4.2.4 LTS update isn't out yet. Closed, and sorry for the noise.
